### PR TITLE
Added more YCbCr validation

### PR DESCRIPTION
**Pull request by sfricke-samsung**
Adds supports for

- VUID-VkImageCreateInfo-format-02653
- VUID-VkImageCreateInfo-format-02564
- VUID-VkImageCreateInfo-format-02561
- VUID-VkImageCreateInfo-format-02562
- VUID-VkImageCreateInfo-format-02563

### CODE REVIEW

**Comment by sfricke-samsung** [comment]
Yes, fixed PR message... the `0256*` and `0265*` got me mixed up several times lol

**Comment by mark-lunarg** [comment]
LGTM, running through CI.

**Comment by mark-lunarg** [comment]
@sfricke-samsung, the new test appears to fail on Nvidia Windows and Linux platforms when rebased on Master.  Can you see if you can reproduce?

**Comment by sfricke-samsung** [comment]
@mark-lunarg think I fixed it, basically the issue was I had `VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_SPEC_VERSION` which I found is `2` but most implementations seems to only support `1` (2 just allows the ability to call GetPDProps earlier which is not needed)

When changing that to 1 it wouldn't get the "extensions not supported, skipping" and with that realized I needed to add

```
vk::GetPhysicalDeviceImageFormatProperties();
if ((image_format_props.sampleCounts & VK_SAMPLE_COUNT_4_BIT) == 0) {
    m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, "VUID-VkImageCreateInfo-samples-02258");
}
```

Since this seems like such a small case to add the injected simulation properties logic for a single test in the set

**Comment by mark-lunarg** [comment]
Hey, @Tony-LunarG, can you take a quick look at the new tests and make sure you're happy with everything?

File: layers/buffer_validation.cpp [hunk 1]
```diff
@@ -1622,6 +1613,38 @@ bool CoreChecks::PreCallValidateCreateImage(VkDevice device, const VkImageCreate
         }
     }
 
+    // Tests for "Formats requiring sampler Y’CBCR conversion for VK_IMAGE_ASPECT_COLOR_BIT image views"
+    if (FormatRequiresYcbcrConversion(pCreateInfo->format)) {
+        if (!enabled_features.ycbcr_image_array_features.ycbcrImageArrays && pCreateInfo->arrayLayers != 1) {
+            const char *error_vuid = (device_extensions.vk_ext_ycbcr_image_arrays) ? "VUID-VkImageCreateInfo-format-02653"
+                                                                                   : "VUID-VkImageCreateInfo-format-02564";
+            skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT, 0, error_vuid,
+                            "vkCreateImage(): arrayLayers=%d and must only be 1 if ycbcrImageArrays feature is not enabled",
```
Comment (Tony-LunarG) [inline layers/buffer_validation.cpp]: This seems inversely worded to me.  I'd suggest something more along the lines of the wording in the spec.  Something like "arrayLayers = %d, but when the ycbcrImagesArrays feature is not enabled, arrayLayers must be 1".

File: layers/buffer_validation.cpp [hunk 2]
```diff
@@ -1622,6 +1613,38 @@ bool CoreChecks::PreCallValidateCreateImage(VkDevice device, const VkImageCreate
...
+                            pCreateInfo->arrayLayers);
+        }
+
+        if (pCreateInfo->mipLevels != 1) {
+            skip |= log_msg(report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_EXT, 0,
+                            "VUID-VkImageCreateInfo-format-02561",
+                            "vkCreateImage(): mipLevels=%d and must only be 1 if using a YCbCr Conversion format",
```
Comment (Tony-LunarG) [inline layers/buffer_validation.cpp]: Suggest "mipLevels = %d, but when using a YCbCr Conversion format,, arrayLayers must be 1".  Similar suggestions fo rthe next two messages.

File: tests/vklayertests_buffer_image_memory_sampler.cpp [hunk 3]
```diff
@@ -8007,13 +8015,48 @@ TEST_F(VkLayerTest, CreateImageYcbcrArrayLayers) {
...
+        m_errorMonitor->SetDesiredFailureMsg(VK_DEBUG_REPORT_ERROR_BIT_EXT, "VUID-VkImageCreateInfo-samples-02258");
+    }
+    CreateImageTest(*this, &image_create_info, "VUID-VkImageCreateInfo-format-02562");
+    image_create_info.samples = VK_SAMPLE_COUNT_1_BIT;
+
+    // invalid imageType
+    image_create_info.extent.height = 1;  // to satisfy 1D requriments
+    image_create_info.imageType = VK_IMAGE_TYPE_1D;
+    CreateImageTest(*this, &image_create_info, "VUID-VkImageCreateInfo-format-02563");
+    image_create_info.imageType = VK_IMAGE_TYPE_2D;
```
Comment (Tony-LunarG) [inline tests/vklayertests_buffer_image_memory_sampler.cpp]: What are the last two lines for?  Was there going to be another test?

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/buffer_validation.cpp]: actually I would go further now and have it fully clear

"arrayLayers = %d, but when the ycbcrImagesArrays feature is not enabled and a ycbcr format is used, arrayLayers must be 1"

File: tests/vklayertests_buffer_image_memory_sampler.cpp (hunk 3 above)
Comment (sfricke-samsung) [reply tests/vklayertests_buffer_image_memory_sampler.cpp]: well more forward-thinking if someone added another test and didn't catch I changed it prior

File: layers/buffer_validation.cpp (hunk 1 above)
Comment (Tony-LunarG) [reply layers/buffer_validation.cpp]: Even better

**Comment by mark-lunarg** [comment]
Running through CI, thanks @Tony-LunarG  for the review.

**Comment by mark-lunarg** [comment]
Thanks @sfricke-samsung, and keep 'em coming!

**Pull request merged**